pkg/doc/util/signature/signer.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package signature
+
+import (
+	"crypto/ed25519"
+	"errors"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/util/signature/internal"
+
+	cryptoapi "github.com/hyperledger/aries-framework-go/pkg/crypto"
+	kmsapi "github.com/hyperledger/aries-framework-go/pkg/kms"
+)
+
+// NewCryptoSigner creates a new signer based on crypto if possible.
+func NewCryptoSigner(crypto cryptoapi.Crypto, kms kmsapi.KeyManager, keyType kmsapi.KeyType) (Signer, error) {
+	switch keyType {
+	case kmsapi.ECDSAP256TypeDER, kmsapi.ECDSAP384TypeDER, kmsapi.ECDSAP521TypeDER,
+		kmsapi.ECDSAP256TypeIEEEP1363, kmsapi.ECDSAP521TypeIEEEP1363,
+		kmsapi.ED25519Type:
+		return internal.NewCryptoSigner(crypto, kms, keyType)
+
+	case kmsapi.ECDSAP384TypeIEEEP1363:
+		// TODO use NewCryptoSigner(crypto, kms, keyType), it's now failing with "invalid signature" error for some reason.
```

Comment:
?? I wonder how P256 and P521 above are passing but not P384?

Reply:
I have no idea. I will check this separately.

Reply:
@kdimak can you please add the issue number in the comment?

Reply:
Fixed in this PR. @llorllale @Baha-sk

Reply:
thank you @kdimak